tuna/miopen/scripts/report.py

```diff
@@ -0,0 +1,251 @@
+  LOGGER.info("configs with faster kernel_time: %f %%", round(prct_negative, 4))
+  LOGGER.info("configs with equal kernel_time: %f %%", round(prct_equal, 4))
+  LOGGER.info("configs with slower kernel_time: %f %% \n",
+              round(prct_positive, 4))
+
+  #averages
+  avg_positive = (dfr['diff'] > 0).mean()
+  avg_negative = (dfr['diff'] < 0).mean()
+
+  LOGGER.info("Mean for configs with faster kernel_time: %s %%", avg_negative)
+  LOGGER.info("Mean for configs with slower kernel_time: %s %%", avg_positive)
+  LOGGER.info("Mean for all configs: %s %%", dfr['diff'].mean())
+
+  dfr['%speedup'] = (dfr['ktime_x'] - dfr['ktime_y']) / (
+      (dfr['ktime_x'] + dfr['ktime_y']) / 2) * 100
+  LOGGER.info("Overall speed-up: %s %%", round(dfr['%speedup'].mean(), 4))
+
+  config_data = f"config_data_sess{args.session_id}_gv{args.golden_v}.csv"
+  LOGGER.info("Config report has been written to file: %s\n", config_data)
+  dfr.to_csv(config_data)
+
+
+def solver_report(args, dfr):
+  """Print detailed tuning analysis"""
+
+  LOGGER.info('Detailed solver report:')
+
+  #dataframe with fastest solvers(solver_x, solver_y)
+  df_compare = dfr.replace(-1, np.nan).groupby('config')[['ktime_x',
+                                                          'ktime_y']].idxmin()
+  df_compare.columns = ['idx_x', 'idx_y']
+
+  df_compare['solver_x'] = df_compare['idx_x'].apply(dfr['solver'].get)
+  df_compare['solver_y'] = df_compare['idx_y'].apply(dfr['solver'].get)
+
+  df_compare['ktime_x'] = df_compare['idx_x'].apply(dfr['ktime_x'].get)
+  df_compare['ktime_y'] = df_compare['idx_y'].apply(dfr['ktime_y'].get)
+
+  #dataframe where solvers have changed
+  dfr_diff_solvers = df_compare.loc[df_compare['solver_x'].ne(
+      df_compare['solver_y'])]
+  #dataframe where solvers are the same
+  dfr_same_solvers = df_compare.loc[df_compare['solver_x'].eq(
+      df_compare['solver_y'])]
+  dfr_same_solvers = dfr_same_solvers.drop(columns=['idx_x', 'idx_y'])
+  dfr_same_solvers['%diff'] = (dfr_same_solvers['ktime_x'] - dfr_same_solvers['ktime_y'])\
+                    / ((dfr_same_solvers['ktime_x'] + dfr_same_solvers['ktime_x'])/2) * 100
+  _, id_solver_map = get_id_solvers()
+  dfr_same_solvers['solver_x'] = dfr_same_solvers['solver_x'].apply(
+      id_solver_map.get)
+  dfr_same_solvers['solver_y'] = dfr_same_solvers['solver_y'].apply(
+      id_solver_map.get)
+  LOGGER.info('Mean %%change for same solver: %s',
+              dfr_same_solvers.loc[:, '%diff'].mean())
+  report_file = f"same_solvers_report_sess{args.session_id}_gv{args.golden_v}.csv"
+  LOGGER.info("Same solvers detailed report has been written to file: %s",
+              report_file)
+  dfr_same_solvers.to_csv(report_file)
+
+  #Percentage difference formula
+  diff_mean = dfr_diff_solvers.groupby(['solver_x', 'solver_y'])\
+                  .apply(lambda grp: ((grp['ktime_x'] - grp['ktime_y'])
+                    / ((grp['ktime_x'] + grp['ktime_x'])/2) * 100 ).mean())
```

Comment:
Note that  (a+a)/2 == 2a/2 == a . So, just do
```
  diff_mean = dfr_diff_solvers.groupby(['solver_x', 'solver_y'])\
                  .apply(lambda grp: ((grp['ktime_x'] - grp['ktime_y'])
                    / grp['ktime_x'] * 100 ).mean())
```

Also, instead of saying "mean change", I'd say use if-else to report "mean slowdown"/"mean speedup"

Reply:
i think we actually want to see the overall mean change